### ISSUE TITLE
GML3: Support writing features with multiple geometry fields

### ISSUE THREAD

**Issue by awaterme** [openlayers/ol3#3328]
`ol.format.GML3.prototype.writeFeatureElement` currently treats only one field as geometry when serializing objects to GML: The field corresponding to `Feature.getGeometryName()`.

The serialization of further geometries of a feature is broken and results in invalid GML (containing "[Object object]" as serialization result).

I suggest to consider not only the geometry-field but also other fields being of type `ol.geom.Geometry`.

The solution is very simple:
```javascript
ol.format.GML3.prototype.writeFeatureElement =
    function(node, feature, objectStack) {
        var fid = feature.getId();
        if (goog.isDef(fid)) {
            node.setAttribute('fid', fid);
        }
        var context = objectStack[objectStack.length - 1];
        goog.asserts.assert(goog.isObject(context));
        var featureNS = goog.object.get(context, 'featureNS');
        var geometryName = feature.getGeometryName();
        if (!goog.isDef(context.serializers)) {
            context.serializers = {};
            context.serializers[featureNS] = {};
        }
        var properties = feature.getProperties();
        var keys = [], values = [];
        for (var key in properties) { // jshint ignore:line
            var value = properties[key];
            if (!goog.isNull(value)) {
                keys.push(key);
                values.push(value);
// --------- modification start: also check type of value -------------- //
               //original:  if (key === geometryName) {             
               if (key === geometryName || value instanceof ol.geom.Geometry) {
// --------- modification end -------------- //
                    if (!(key in context.serializers[featureNS])) {
                        context.serializers[featureNS][key] = ol.xml.makeChildAppender(
                            this.writeGeometryElement, this);
                    }
                } else {
                    if (!(key in context.serializers[featureNS])) {
                        context.serializers[featureNS][key] = ol.xml.makeChildAppender(
                            ol.format.XSD.writeStringTextNode);
                    }
                }
            }
        }
        var item = goog.object.clone(context);
        item.node = node;
        ol.xml.pushSerializeAndPop(/** @type {ol.xml.NodeStackItem} */
            (item), context.serializers,
            ol.xml.makeSimpleNodeFactory(undefined, featureNS),
            values,
            objectStack, keys);
    };
```

**Comment by bartvde** [comment]
can you create a PR ? TIA

**Comment by probins** [comment]
I think this was fixed with #4237

**Comment by ahocevar** [comment]
Fixed with #4327.
